### PR TITLE
016/create global config interfae

### PR DESCRIPTION
**Pull request by mikeerickson**
I have implemented a number of items across this PR, but I want you to have at what I have done as it doesn't completely align with your RFC outlines #19 and #20 

In this PR (which i can see changing after reading your comments) addresses

- global `drome.config.js` file (locate in users home directory)
- extended tasks (merged with existing drome config at runtime)
- shows help and version info (from cli)

Please have a look at this PR and let me know your thoughts, then I will have a better feel for your requirements and adjust code accordingly.

### CODE REVIEW

File: lib/task.js [hunk 1]
```diff
@@ -9,32 +11,69 @@ async function asyncForEach(array, callback) {
 }
 
 const run = async (task, name, cliOptions = []) => {
+    // default to help if nothing is passed to drome
+    let cmd = (typeof name === 'undefined') ? 'help' : '';
+    if (typeof task === 'undefined') {
+        cmd = hasOption(name, ['-H', '-h', '--help', 'help']) ? 'help' : cmd;
```
[the remark this answers is not in the document]
Comment (mikeerickson) [reply lib/task.js]: @kodipe I am now working on this format `drome run <task>` as I agree it is a better syntax

File: lib/drome.js [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+const { hasOption } = require('./cliUtils');
 const { getObjectFromPath } = require('./utils');
 
 const drome = (config, start, cliOptions) => {
     let conf = config();
+
+    let silent = hasOption(cliOptions, ['--silent', '-s', '-S', 'silent']);
+    silent = (start === 'help' || start === 'version'); // if we are using cli command, suppress logging
+
+    !silent && logger.header('==> staring process');
```
[the remark this answers is not in the document]
Comment (mikeerickson) [reply lib/drome.js]: @kodipe Darn typos :-)

File: lib/task.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikeerickson) [reply lib/task.js]: @kodipe I am going to assume that issuing `drome run` without supplying the task will run all tasks (as it currently does when using `drome` alone)

[the remark this answers is not in the document]
Comment (kodipe) [reply lib/task.js]: @mikeerickson exactly :)

**Pull request merged**